java/runtime/src/main/java/org/ray/runtime/DefaultRayRuntimeFactory.java

```diff
@@ -25,9 +27,17 @@ public RayRuntime createRayRuntime() {
       if (rayConfig.runMode == RunMode.SINGLE_PROCESS) {
         runtime = new RayDevRuntime(rayConfig, functionManager);
       } else {
+        RunManager manager = null;
+        if (rayConfig.getRedisAddress() == null) {
+          manager = new RunManager(rayConfig);
+          manager.startRayProcesses(true);
+        }
+        RayNativeRuntime.setup(rayConfig);
```

Comment:
why move RunManager outside?

Reply:
The `RunManager` should be executed before `setup` since `RunManager` will prepare `session_dir`.